addons/website/static/src/components/fields/publish_button.js

```diff
@@ -0,0 +1,23 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+
+const { Component } = owl;
+
+class PublishField extends Component {
+    get hover() {
+        return this.props.value ? this.env._t("Unpublish") : this.env._t("Publish");
+    }
+    get hoverClass() {
+        return 'o_stat_text o_hover ' + (this.props.value ? 'text-danger' : 'text-success');
+    }
+    get value() {
+        return this.props.value ? this.env._t("Published") : this.env._t("Unpublished");
+    }
+    get valueClass() {
+        return 'o_stat_text o_not_hover ' + (this.props.value ? 'text-success' : 'text-danger');
+    }
```

Comment:
Is any of this necessary to do in JS? Why is this not in the XML template directly?